cmd/geth/shardingcmd.go

```diff
@@ -1,46 +1,59 @@
+	"github.com/ethereum/go-ethereum/sharding/notary"
+	"github.com/ethereum/go-ethereum/sharding/proposer"
 	cli "gopkg.in/urfave/cli.v1"
 )
 
 var (
-	notaryClientCommand = cli.Command{
-		Action:    utils.MigrateFlags(notaryClient),
-		Name:      "sharding-notary",
-		Aliases:   []string{"shard-notary"},
-		Usage:     "Start a sharding notary client",
+	shardingCommand = cli.Command{
+		Action:    utils.MigrateFlags(shardingClient),
+		Name:      "sharding",
+		Usage:     "Start a sharding-enabled node",
 		ArgsUsage: "[endpoint]",
-		Flags:     []cli.Flag{utils.DataDirFlag, utils.PasswordFileFlag, utils.NetworkIdFlag, utils.IPCPathFlag, utils.DepositFlag},
+		Flags:     []cli.Flag{utils.ProtocolFlag, utils.DataDirFlag, utils.PasswordFileFlag, utils.NetworkIdFlag, utils.IPCPathFlag, utils.DepositFlag},
 		Category:  "SHARDING COMMANDS",
 		Description: `
-Launches a sharding notary client that connects to a running geth node and submit collations to a Sharding Manager Contract. This feature is a work in progress.
-`,
-	}
-	proposerClientCommand = cli.Command{
-		Action:    utils.MigrateFlags(proposerClient),
-		Name:      "sharding-proposer",
-		Aliases:   []string{"shard-proposer"},
-		Usage:     "Start a sharding proposer client",
-		ArgsUsage: "[endpoint]",
-		Flags:     []cli.Flag{utils.DataDirFlag, utils.PasswordFileFlag, utils.NetworkIdFlag, utils.IPCPathFlag},
-		Category:  "SHARDING COMMANDS",
-		Description: `
-Launches a sharding proposer client that connects to a running geth node and proposes collations to notary node. This feature is a work in progress.
+Launches a sharding node that manages services related to submitting collations to a Sharding Manager Contract, notary and proposer services, and shardp2p connections. This feature is a work in progress.
 `,
 	}
 )
 
-func notaryClient(ctx *cli.Context) error {
-	c := notary.NewNotary(ctx)
-	return c.Start()
+func shardingClient(ctx *cli.Context) error {
+	// configures a sharding-enabled node using the cli's context.
+	shardingNode := node.NewNode(ctx)
+	registerShardingServices(shardingNode)
+	defer shardingNode.Close()
+	// starts a connection to a geth node and kicks off every registered service.
+	return shardingNode.Start()
 }
 
-func proposerClient(ctx *cli.Context) error {
-	p := proposer.NewProposer(ctx)
-	return p.Start()
+// registerShardingServices sets up either a notary or proposer
+// sharding service dependent on the ClientType cli flag. We should be defining
+// the services we want to register here, as this is the geth command entry point
+// for sharding.
+func registerShardingServices(n node.Node) error {
+	protocolFlag := n.Context().GlobalString(utils.ProtocolFlag.Name)
+
+	err := n.Register(func(ctx *cli.Context) (sharding.Service, error) {
+		// TODO(terenc3t): handle case when we just want to start an observer node.
+		if protocolFlag == "notary" {
+			return notary.NewNotary(n.Context(), n)
+		}
+		return proposer.NewProposer(n.Context(), n)
```

Comment:
if (protocolFlag == "proposer") ? 

The default behavior would be just an observer right?

Reply:
Yeah, added a TODO for default behavior assigned to @terenc3t if the flag is not passed in